pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -92,6 +92,18 @@
      */
     long getEventTime();
 
+    /**
+     * Get the sequence id associated with this message. It is typically set by the applications via
+     * {@link MessageBuilder#setSequenceId(long)}.
+     *
+     * <p>If there isn't any sequence id associated with this event, it will return -1.
+     *
+     * @return sequence id associated with this message.
+     * @see MessageBuilder#setEventTime(long)
+     * @since 1.22.0
+     */
+    long getSequenceId();
```

Comment:
Since the sequence id is associated with a particular producer, I think we should also expose the `producerName`

Reply:
good point. will add

Reply:
done - exposed producer name